xlators/mgmt/glusterd/src/glusterd-svc-mgmt.c

```diff
@@ -517,3 +517,43 @@ glusterd_muxsvc_conn_init(glusterd_conn_t *conn, glusterd_svc_proc_t *mux_proc,
     }
     return ret;
 }
+
+/*
+ * A generic function replacing two functions,
+ * glusterd_bitdsvc_start and glusterd_scrubsvc_start
+ * wherein both do the same set of operations.
+ */
+
+int
+glusterd_genericsvc_start(glusterd_svc_t *svc, int flags)
+{
+    int i = 0;
+    int ret = -1;
+    dict_t *cmdline = NULL;
+    char key[16] = {0};
+    char *options[] = {svc->name, "--process-name", NULL};
+
+    cmdline = dict_new();
+    if (!cmdline) {
+        gf_smsg(THIS->name, GF_LOG_ERROR, errno, GD_MSG_DICT_CREATE_FAIL, NULL);
+        goto error_return;
+    }
+
+    for (i = 0; options[i]; i++) {
+        ret = snprintf(key, sizeof(key), "arg%d", i);
```

Comment:
nit: a fail check after this line could be helpful?

Reply:
If the check is for overflow, I think we can ignore it as the size of data being stored in key would be safely within the limits according to the flow of the code here.